Allow system_dbusd_t to start/stop all units

dsugar100 opened the pull request:
Examples of denials I'm seeing requiring this type of access: 

node=localhost type=USER_AVC msg=audit(1689811749.504:399): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='avc:  denied  { start } for auid=n/a uid=0 gid=81 path="/usr/lib/systemd/system/systemd-hostnamed.service" cmdline="/usr/bin/dbus-broker-launch --scope system --audit" function="bus_unit_method_start_generic" scontext=system_u:system_r:system_dbusd_t:s0 tcontext=system_u:object_r:systemd_unit_t:s0 tclass=service permissive=0 exe="/usr/lib/systemd/systemd" sauid=0 hostname=? addr=?  terminal=?'␝UID="root" AUID="unset" AUID="root" UID="root" GID="dbus" SAUID="root" 
node=localhost type=USER_AVC msg=audit(1692287535.229:262): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='avc:  denied  { start } for auid=n/a uid=0 gid=81 path="/usr/lib/systemd/system/systemd-logind.service" cmdline="/usr/bin/dbus-broker-launch --scope system --audit" function="bus_unit_method_start_generic" scontext=system_u:system_r:system_dbusd_t:s0 tcontext=system_u:object_r:systemd_unit_t:s0 tclass=service permissive=1 exe="/usr/lib/systemd/systemd" sauid=0 hostname=? addr=? terminal=?'␝UID="root" AUID="unset" AUID="root" UID="root" GID="dbus" SAUID="root" 
node=localhost type=USER_AVC msg=audit(1692305808.055:375): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='avc:  denied  { start } for auid=n/a uid=0 gid=81 path="/usr/lib/systemd/system/accounts-daemon.service" cmdline="/usr/bin/dbus-broker-launch --scope system --audit" function="bus_unit_method_start_generic" scontext=system_u:system_r:system_dbusd_t:s0 tcontext=system_u:object_r:systemd_unit_t:s0 tclass=service permissive=0 exe="/usr/lib/systemd/systemd" sauid=0 hostname=? addr=?  terminal=?' UID="root" AUID="unset" AUID="root" UID="root" GID="dbus" SAUID="root"

gtrentalancia commented:
These permissions (and the ones above them) are only needed when using systemd. So, I suppose you might want to enclose them in an ifdef init_systemd block...

pebenito reviewed (changes requested):
Please put in the systemd block.

dsugar100 commented:
I moved just the two lines I added.  I wasn't sure what else of those are only needed for systemd.   I might assume 'init_start_system' could be.  But, the rest I really wasn't sure about.  I can move more in the ifdef if desired.

pebenito commented:
> I moved just the two lines I added. I wasn't sure what else of those are only needed for systemd. I might assume 'init_start_system' could be. But, the rest I really wasn't sure about. I can move more in the ifdef if desired.

Thanks. I think `init_start_system(system_dbusd_t) # needed by dbus-broker` can also move, as that has an obvious systemd-only perm. If there's anything else, it is unclear from the policy alone.

gtrentalancia commented:
Yes, init_start_system(system_dbusd_t) belongs to systemd policy.

gtrentalancia commented:
Thanks.
